WooCommerce/Classes/ViewRelated/Coupons/Add and Edit Coupons/CouponExpiryDateView.swift

```diff
@@ -16,31 +18,59 @@ struct CouponExpiryDateView: View {
                     DatePicker("Date picker", selection: $date, displayedComponents: .date)
                         .environment(\.timeZone, timezone)
                         .datePickerStyle(GraphicalDatePickerStyle())
-                        .onChange(of: date) { newDate in
-                            completion(newDate)
-                        }
+                        .onChange(of: date, perform: { newDate in
+                                isRemovalEnabled = true
+                            })
                     Spacer()
+                    VStack {
+                        Divider()
+                        Button(action: {
+                            onCompletion(nil)
+                            presentationMode.wrappedValue.dismiss()
+                        }, label: { Text(Localization.removeButton) })
+                        .buttonStyle(PlainButtonStyle())
+                        .padding(.vertical, Constants.margin)
+                        .foregroundColor(Color(.error))
+                        Divider()
+                    }
+                    .padding(.top, Constants.verticalMargin)
+                    .renderedIf(isRemovalEnabled)
                 }
             }
         }
         .ignoresSafeArea(.container, edges: [.horizontal, .bottom])
         .navigationBarTitleDisplayMode(.inline)
         .navigationTitle(Localization.title)
+        .toolbar {
+            ToolbarItem(placement: .confirmationAction) {
+                Button(action: {
+                    onCompletion(date)
+                    presentationMode.wrappedValue.dismiss()
+                }, label: { Text(Localization.doneButton) })
+                        .foregroundColor(Color(.accent))
+            }
+        }
     }
 }
 
 // MARK: - Constants
 //
 private extension CouponExpiryDateView {
+    enum Constants {
+        static let margin: CGFloat = 8
+        static let verticalMargin: CGFloat = 40
+    }
 
     enum Localization {
         static let title = NSLocalizedString("Coupon expiry date",
                                              comment: "Title of the view for selecting an expiry date for a coupon.")
+        static let doneButton = NSLocalizedString("Done", comment: "Button to finish selecting the Coupon expiry date in the AddEditCoupon screen")
+        static let removeButton = NSLocalizedString("Remove Expiry Date", comment: "Button to remove the Coupon expiry date in the AddEditCoupon screen")
```

Comment:
Super nit: we should use "normal language" on the comments of the localized strings since the target readers are translators 😅 So instead of "AddEditCoupon screen" let's say "Add or Edit Coupon screen".

Reply:
Oops 😬, thanks for explaining this context to me! Fixed! https://github.com/woocommerce/woocommerce-ios/pull/6953/commits/096813bed4f06363d528e31002b1cc3db673fb2e